docker/puppetdb/docker-entrypoint.d/20-wait-for-hosts.sh

```diff
@@ -24,13 +24,21 @@ wait_for_host() {
   fi
 }
 
+wait_for_host_port() {
+  /wtfc.sh --timeout=$PUPPETDB_WAITFORHOST_SECONDS --interval=1 --progress "curl --silent --fail --connect-timeout 1 --insecure '${1}:${2}'"
+  if [ $? -ne 0 ]; then
+    error "host $1:$2 does not appear to be listening"
+  fi
+}
+
 PUPPETDB_WAITFORHOST_SECONDS=${PUPPETDB_WAITFORHOST_SECONDS:-30}
 PUPPETDB_WAITFORHEALTH_SECONDS=${PUPPETDB_WAITFORHEALTH_SECONDS:-600}
 PUPPETSERVER_HOSTNAME="${PUPPETSERVER_HOSTNAME:-puppet}"
 CONSUL_HOSTNAME="${CONSUL_HOSTNAME:-consul}"
 CONSUL_PORT="${CONSUL_PORT:-8500}"
 
 wait_for_host "postgres"
+wait_for_host_port "postgres" 5432
```

Comment:
Do we still need the `wait_for_host "postgres"` above if we're doing this now?

I'm good with this, just wondering.

Reply:
Probably not... but need to understand failure cases for `curl` with DNS resolutions and how that might be different than `ping`